parsec/core/fs/workspacefs/versioning_helpers.py

```diff
@@ -50,109 +51,187 @@ class ManifestDataAndPaths(NamedTuple):
     destination: FsPath
 
 
-async def list_versions(
-    workspacefs, path: FsPath, skip_minimal_sync: bool = True
-) -> List[TimestampBoundedData]:
+class ManifestDataAndMutablePaths:
     """
-    Raises:
-        FSError
-        FSBackendOfflineError
-        FSWorkspaceInMaintenance
-        FSRemoteManifestNotFound
+    Contains a manifest and his known current, previous and next paths.
+    Unlike ManifestDataAndPaths, attributes are mutable.
+    """
+
+    def __init__(self, manifest, source_path=None, destination_path=None, current_path=None):
+        self.manifest = manifest
+        self.source = source_path
+        self.destination = destination_path
+        self.current = current_path
```

Comment:
Maybe use `attrs` for this class, so you have a syntax similar to the other data classes.

Also, those would have been a good use case for [dataclasses](https://docs.python.org/3/library/dataclasses.html) if we were using python 3.7.

Reply:
Done!